src/App.tsx

```diff
@@ -15,48 +17,108 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
-export const App: React.FC = () => {
+const SORT_FIELD_ALPHABET = 'alphabet';
+const SORT_FIELD_LENGTH = 'length';
+
+interface FilterParams {
+  sortField: string;
+  reverse: boolean;
+}
+
+function getPreparedGoods(
+  goods: string[],
+  { sortField, reverse }: FilterParams,
+) {
+  const prepearedGoods = [...goods];
+
+  if (sortField) {
+    prepearedGoods
+      .sort((good1, good2) => {
+        switch (sortField) {
+          case SORT_FIELD_ALPHABET:
+            return good1.localeCompare(good2);
+
+          case SORT_FIELD_LENGTH:
+            return good1.length - good2.length;
+
+          default:
+            return 0;
+        }
+      });
+  }
+
+  if (reverse) {
+    prepearedGoods.reverse();
+  }
+
+  return prepearedGoods;
+}
+
+export const App = () => {
+  const [sortField, setSortField] = useState('');
+  const [reverse, setReverse] = useState(false);
+  const visibleGoods = getPreparedGoods(
+    goodsFromServer,
+    { sortField, reverse },
+  );
+
   return (
     <div className="section content">
       <div className="buttons">
         <button
+          onClick={() => setSortField(SORT_FIELD_ALPHABET)}
           type="button"
-          className="button is-info is-light"
+          className={cn(
+            'button',
+            'is-info',
+            { 'is-light': sortField !== SORT_FIELD_ALPHABET },
+          )}
         >
           Sort alphabetically
         </button>
 
         <button
+          onClick={() => setSortField(SORT_FIELD_LENGTH)}
```

Comment:
You can write handler for it ( not critical )

Reply:
done